bokehjs/src/coffee/models/sources/cds_view.ts

```diff
@@ -76,36 +75,37 @@ export class CDSView extends Model {
       this.indices = intersection.apply(this, indices)
     } else {
       if (this.source instanceof ColumnarDataSource) {
-        this.indices = this.source != null ? this.source.get_indices() : undefined
+        this.indices = this.source.get_indices()
       }
     }
 
-    return this.indices_map_to_subset()
+    this.indices_map_to_subset()
   }
 
-  indices_map_to_subset() {
+  indices_map_to_subset(): void {
     this.indices_map = {}
-    return range(0, this.indices.length).map((i) =>
-      (this.indices_map[this.indices[i]] = i))
+    for (let i = 0; i < this.indices.length; i++){
+      this.indices_map[this.indices[i]] = i
+    }
   }
 
-  convert_selection_from_subset(selection_subset): HitTestResult {
-    const selection_full = create_hit_test_result()
+  convert_selection_from_subset(selection_subset: Selection): Selection {
+    const selection_full = new Selection()
     selection_full.update_through_union(selection_subset)
-    const indices_1d = (selection_subset['1d']['indices'].map((i) => this.indices[i]))
-    selection_full['1d']['indices'] = indices_1d
+    const indices_1d = (selection_subset.indices.map((i) => this.indices[i]))
+    selection_full.indices = indices_1d
     return selection_full
   }
 
-  convert_selection_to_subset(selection_full): HitTestResult {
-    const selection_subset = create_hit_test_result()
+  convert_selection_to_subset(selection_full: Selection): Selection {
+    const selection_subset = new Selection()
     selection_subset.update_through_union(selection_full)
-    const indices_1d = (selection_full['1d']['indices'].map((i) => this.indices_map[i]))
-    selection_subset['1d']['indices'] = indices_1d
+    const indices_1d = (selection_full.indices.map((i) => this.indices_map[i]))
+    selection_subset.indices = indices_1d
     return selection_subset
   }
 
-  convert_indices_from_subset(indices) {
+  convert_indices_from_subset(indices: Number[]) {
```

Comment:
Use lower case types for primitives. In fact, run `gulp scripts:tslint`, it will point out errors of this and other kinds.